app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -35,7 +35,7 @@ def show
     param :product_id, :number, :required => true, :desc => N_("ID of a product to list repository sets from")
     def available_repositories
       scan_cdn = sync_task(::Actions::Katello::RepositorySet::ScanCdn, @product, @product_content.content.id)
-      repos = scan_cdn.output[:results]
+      repos = scan_cdn.output[:results] || []
```

Comment:
If the empty array is returned, will the user still see somewhere that they got a 403 from the CDN?